modyn/selector/internal/selector_strategies/abstract_selection_strategy.py

```diff
@@ -73,6 +64,27 @@ def __init__(self, config: SelectionStrategy, modyn_config: dict, pipeline_id: i
         self._trigger_sample_directory = self._modyn_config["selector"]["trigger_sample_directory"]
         self._storage_backend = self._init_storage_backend()
 
+    def _update_next_trigger_id(self) -> None:
+        tid = threading.get_native_id()
+        pid = os.getpid()
+
+        with MetadataDatabaseConnection(self._modyn_config) as database:
+            last_trigger_id = (
```

Comment:
I propose to use the atomic transactional capabilities of our database to force different processes to use unique `trigger_id`s. With the current version (`select max query`) we could still run into the problem that two parallel processes request a new trigger id at the same time before writing the trigger back to the db. Then both processes would end up with the same trigger id. 
Doing the `self._next_trigger_id = last_trigger_id + 1` logic  in the database is more reliable I think. We could either use an autoincrement table primary key or sth. more lightweight like a postgres series.

Reply:
generally, i agree. we use a mp.manager.Lock() to prevent what you are saying. However of course it's better to use the db here. There is some historic reason we do not rely on the DB. IIRC we somehow needed the actual ID of the next trigger before we insert and at some point, we did not have a composite primary key of (pid, tid) but just tid(), so there was no obvious +1 logic. We should refactor this, but my proposal would be to merge this anyways as a hotfix to have a somewhat functional main branch (the locks at the grpc servicer level should hopefully make it correct)

Reply:
Sounds good :) It's already an improvement from the status quo!